### PR TITLE
Add doc about how to get self subject attributes

### PR DESCRIPTION
**Pull request by nabokihms**
Documentation for:

  - [x] KEP (`k/enhancements`) update PR(s): https://github.com/kubernetes/enhancements/pull/3326
  - [x] Code (`k/k`) update PR(s): https://github.com/kubernetes/kubernetes/pull/111333
  - [ ] Docs (`k/website`) update PR(s): https://github.com/kubernetes/website/pull/35385

### CODE REVIEW

**Comment by divya-mohan0209** [comment]
@nabokihms Hello there! I'm assuming this is for the 1.25 release. If yes, please could you resubmit the PR against the dev-1.25 branch instead of the main branch?

**Comment by cathchu** [comment]
Hey there 👋 @nabokihms , 1.25 Release Docs Shadow here!

Thanks for filing this placeholder. As a reminder, final docs PRs are due August 9th. Take a look at [Documenting a release](https://kubernetes.io/docs/contribute/new-content/new-features/#for-developers-or-other-sig-members) for additional information for the docs requirement for the release.

Thanks again!

**Comment by reylejano** [comment]
/assign @cathchu 
/cc @kcmartin 
/hold
hold on confirming the status of the related enhancement https://github.com/kubernetes/enhancements/issues/3325

**Comment by reylejano** [comment]
/milestone clear
@nabokihms , since this enhancement is no longer tracked for 1.25, you can target this PR for `main` and change the target when the `dev-1.26` branch opens when 1.25 is released

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (Sea-n) [inline content/en/docs/reference/access-authn-authz/authentication.md:1244]:
```suggestion
{{< feature-state for_k8s_version="v1.26" state="alpha" >}}
```

**Review by Sea-n — COMMENTED** [review]
It's [targeting v1.26](https://github.com/kubernetes/enhancements/issues/3325#issuecomment-1234293022) now.

And please change the target branch to `dev-1.26`.

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1242]:
```suggestion
## API access to authentication information for a client {#self-subject-review}
```

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1248]: Does this command also work with control plane versions before v1.26? If not, what happens when you try?

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1282]: Remember to add `APISelfSubjectAttributesReview` into https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1246]:
```suggestion
You can use the `SelfSubjectReview` API, to find out how your Kubernetes cluster maps your authentication
information to identify you as a client. This works whether you are authenticating as a user (typically representing
a real person) or as a ServiceAccount.
```

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1276]: If there are extra data present, do these show up? If so, maybe we need a simple example and a more complicated example.

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (nabokihms) [reply content/en/docs/reference/access-authn-authz/authentication.md]: Fixed this one

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (nabokihms) [reply content/en/docs/reference/access-authn-authz/authentication.md:1328]: Added :+1:

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (nabokihms) [reply content/en/docs/reference/access-authn-authz/authentication.md]: I applied your suggestion.

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (nabokihms) [reply content/en/docs/reference/access-authn-authz/authentication.md]: I added a simple and a complicated example.

**Comment by nabokihms** [comment]
@sftim Hello, I fixed the text according to you comments

- [ ] I would update https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/ to mention that you can use kubectl auth who-am-i, but only if your cluster has the feature enabled.

- [ ] We should document the format of a SelfSubjectReview and its fields (in https://k8s.io/docs/reference/access-authn-authz/authentication/ )

- [ ] We should document whether access mechanisms apply. Can everyone create a SelfSubjectReview, or do they need RBAC permissions on selfsubjectreviews before they can?

- [ ] How does this API interact with [impersonation](https://kubernetes.io/docs/reference/access-authn-authz/authentication/)?

- [ ] Can we mention when this API is especially useful (eg, in cluster that uses a custom authz webhook, or a cluster that uses an aggregated API that enforces additional authz).

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (nabokihms) [reply content/en/docs/reference/access-authn-authz/authentication.md:1279]: This is a copypast from the KEP where you also requested to fix the same line. Sorry. I will fix this.

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (nabokihms) [reply content/en/docs/reference/access-authn-authz/authentication.md:1276]: Yes, I will.

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (nabokihms) [reply content/en/docs/reference/access-authn-authz/authentication.md:1289]: I split this into two sections: Simple / Complex and JSON / YAML. The first section is just a list, the second keeps the tabs formatting.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (nabokihms) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:61]: My bad, fixed.

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (nabokihms) [reply content/en/docs/reference/access-authn-authz/authentication.md:1247]: Added your variant.

**Comment by nabokihms** [comment]
Fixed the doc according to all suggestions, thanks! Also added a not to the https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/ .

**Comment by sftim** [comment]
New motivation for a hold: we should wait until the k/k PR actually merges.

@nabokihms what PR is that? (please link to this in the description for _this_ PR).

/hold

File: content/en/docs/tasks/access-application-cluster/configure-access-multiple-clusters.md
Comment (sftim) [inline content/en/docs/tasks/access-application-cluster/configure-access-multiple-clusters.md:392]:
```suggestion
## Check the subject represented by the kubeconfig
```

File: content/en/docs/tasks/access-application-cluster/configure-access-multiple-clusters.md
Comment (sftim) [inline content/en/docs/tasks/access-application-cluster/configure-access-multiple-clusters.md:399]:
```suggestion
Read [API access to authentication information for a client](/docs/reference/access-authn-authz/authentication/#self-subject-review)
to learn about this in more detail.
```

**Review by sftim — COMMENTED** [review]
A couple of suggestions.

**Comment by nabokihms** [comment]
@sftim I updated the description (previously, there was a link to KEP with the same list of PRs).

The feature is merged into the main branch.

**Comment by sftim** [comment]
/hold cancel

I recommend [squashing the commits](https://www.k8s.dev/docs/guide/github-workflow/#squash-commits) down to 1.

**Comment by nabokihms** [comment]
@liggitt @sftim, is there anything left? Are we waiting for the review from one of the sig members?

**Comment by liggitt** [comment]
technical content lgtm

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1362]: nits:
```suggestion
for example, if you use [webhook token authentication](/docs/reference/access-authn-authz/authentication/#webhook-token-authentication) or [authenticating proxy](/docs/reference/access-authn-authz/authentication/#authenticating-proxy).
```

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1366]: nit:
```suggestion
including [impersonation](/docs/reference/access-authn-authz/authentication/#user-impersonation).
```

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1378]:
```suggestion
* the API server for your cluster has the `authentication.k8s.io/v1alpha1`
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:637]:
```suggestion
- `APISelfSubjectAttributesReview`: Activate the `SelfSubjectReview` API which allows users
  to see the requesting subject's authentication information.
  See [API access to authentication information for a client](/docs/reference/access-authn-authz/authentication/#self-subject-review)
  for more details.
```

File: content/en/docs/tasks/access-application-cluster/configure-access-multiple-clusters.md
Comment (sftim) [inline content/en/docs/tasks/access-application-cluster/configure-access-multiple-clusters.md:397]:
```suggestion
There is a `kubectl` alpha subcommand command to check subject attributes, such as username,
for your selected Kubernetes client context: `kubectl alpha auth whoami`.
```

**Review by sftim — COMMENTED** [review]
/lgtm

with a few nits that you can fix if you wish to @nabokihms

**Comment by nabokihms** [comment]
@sftim Thank you for your suggestions. I applied them all.

**Review by krol3 — APPROVED** [review]
LGTM cc @mickeyboxell

File: content/en/docs/reference/access-authn-authz/authentication.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/authentication.md:1258]: (nit; not a release blocker for alpha)
```suggestion
        "provider_id": ["token.company.example"]
```

`example` is a documentation top level domain; dev is a [real domain](https://get.dev/).

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:381]: Should there be a line after this one?

**Review by sftim — COMMENTED** [review]
/lgtm cancel

One thing to fix: the feature gate description looks truncated.

**Comment by natalisucks** [comment]
@nabokihms Hello from the SIG Auth triage meeting! There's a nit above from Tim that would be great to get fixed before we merge; I'm happy to be the one to re-add an LGTM when the above is looked over 👍 Thanks so much!

**Comment by nabokihms** [comment]
@sftim @natalisucks, sorry for the delay. I applied all suggestions.

**Comment by krol3** [comment]
Hi @nabokihms, This PR needs a doc review by **Mon Nov 28th** to get this into the release. Please reach out to required SIGs to get their review. Thank you!
cc: @sftim

**Pull request merged**
